kvdb/src/lib.rs

```diff
@@ -129,3 +137,33 @@ pub trait KeyValueDB: Sync + Send + parity_util_mem::MallocSizeOf {
 		IoStats::empty()
 	}
 }
+
+/// Return for a start inclusive prefix, the non inclusive end.
+/// This assume key are ordered in a lexicographical order.
+pub fn end_prefix(prefix: &[u8]) -> Vec<u8> {
+	let mut end_range = prefix.to_vec();
+	while let Some(0xff) = end_range.last() {
+		end_range.pop();
+	}
+	if let Some(byte) = end_range.last_mut() {
+		*byte += 1;
+	}
+	end_range
+}
+
+#[cfg(test)]
+mod test {
+	use super::end_prefix;
+
+	#[test]
+	fn end_prefix_test() {
+		assert_eq!(end_prefix(&[5, 6, 7]), vec![5, 6, 8]);
+		assert_eq!(end_prefix(&[5, 6, 255]), vec![5, 7]);
+		// this is incorrect as the result is before start
```

Comment:
In what sense is it incorrect? Is it "misusage"? Or is the `end_prefix()` handling this incorrectly?

Reply:
I also thought this comment was a bit confusing, but the test below uses `assert not equal`, which maybe redundant in the presence of `assert_eq` below that.

Reply:
Yes it is redundant, it is mostly documentation (at some point I was not remembering why we can recurse on popping the 0xff).